packages/nouns-contracts/contracts/libs/Inflate.sol

```diff
@@ -0,0 +1,870 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0 <0.9.0;
+
+/// @notice Based on https://github.com/madler/zlib/blob/master/contrib/puff
+/// @dev Modified the original code for gas optimizations
+/// 1. Disable overflow/underflow checks
+/// 2. Chunk some loop iterations
+library Inflate {
```

Comment:
Spent an hour playing with additional gas optimizations. May not be worth the complexity, but reduces `tokenURI` by 6 million gas or so:

https://gist.github.com/solimander/9045a13be779fcfb50adca6595828522 (Some TODOs in there)

Reply:
this is awesome, thank you!
if we feel comfortable deploying Inflate as an upgradable dependency, it seems best to ship these optimizations as a separate project, rather than trying to fit them into our current timeline.

@solimander @davidbrai wdyt?

Reply:
@eladmallel Yes, sounds good 👍